sql/table.go

```diff
@@ -155,17 +155,19 @@ func encodeIndexKey(index structured.IndexDescriptor,
 	var key []byte
 	key = append(key, indexKey...)
 
-	for i, id := range index.ColumnIDs {
-		j, ok := colMap[id]
-		if !ok {
-			return nil, fmt.Errorf("missing %q primary key column", index.ColumnNames[i])
-		}
-		// TOOD(pmattis): Need to convert the values[i] value to the type expected by
-		// the column.
+	for _, id := range index.ColumnIDs {
 		var err error
-		key, err = encodeTableKey(key, values[j])
-		if err != nil {
-			return nil, err
+
+		if j, ok := colMap[id]; ok {
+			key, err = encodeTableKey(key, values[j])
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			key, err = encodeTableKey(key, parser.DNull{})
+			if err != nil {
+				return nil, err
+			}
```

Comment:
A preferred alternative is:

// pick value or Null
var val parser.Datum
if j, ok := colMap[id]; ok {
val = values[j]
} else {
val = parser.DNull{}
}
var err error
if key, err = encodeTableKey(key, val); err != nil {
return nil, err
}